malaria-ios/ViewControllers/UserProfileViewController.swift

```diff
@@ -0,0 +1,199 @@
+import UIKit
+import DoneToolbarSwift
+
+// MARK: - Protocol
+
+protocol SaveRemainingPillsProtocol: class {
+  func saveRemainingPills(textField: UITextField)
+}
+
+// MARK: - Table View controller
+
+class UserProfileViewController: UIViewController {
+  
+  @IBOutlet weak var remindMeWeeksButton: UIButton!
+  @IBOutlet weak var tableView: UITableView!
+  @IBOutlet weak var tableViewHeightConstraint: NSLayoutConstraint!
+  @IBOutlet weak var remainingLabel: UILabel!
+  
+  private var medicine: [Medicine] = []
+  private var reminderValue: PillStatusNotificationsManager.ReminderInterval = .OneWeek {
+    didSet {
+      remindMeWeeksButton.setTitle(reminderValue.toString(), forState: .Normal)
+    }
+  }
+  
+  // Core Data
+  private var context: NSManagedObjectContext?
+  private var medicineManager: MedicineManager?
+  private var psnm: PillStatusNotificationsManager?
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    // Set background image
+    view.backgroundColor = UIColor(patternImage: UIImage(named: "background")!)
+    
+    // Get the last reminder value from the User Defaults
+    let value = UserSettingsManager.UserSetting.PillReminderValue.getString()
+    
+    remindMeWeeksButton.setTitle(value, forState: .Normal)
+
+    self.reminderValue = value.toReminderValue()
+  }
+  
+  override func viewWillAppear(animated: Bool) {
+    super.viewWillAppear(animated)
+    
+    context = CoreDataHelper.sharedInstance.createBackgroundContext()!
+    medicineManager = MedicineManager(context: context!)
+    psnm = PillStatusNotificationsManager(context: context!)
+    
+    getRegisteredMedicine()
+    calculateMedicineLeft()
+    recalculateTableHeight()
+  }
+  
+  func dismissKeyboard() {
+    view.endEditing(true)
+  }
+  
+  func getRegisteredMedicine() {
+    medicine = []
+    medicine.append(medicineManager!.getCurrentMedicine()!)
+    
+    // To get all medicine use only:
+    // medicine = medicineManager!.getRegisteredMedicines()
```

Comment:
The code is clear on this. I understand what you did here. I suggest remove this comment and document on the official documentation